charts/fluid-datamigrate/juicefs/templates/statefulset.yaml

```diff
@@ -0,0 +1,76 @@
+{{- if gt (.Values.datamigrate.parallelism | int) 1 }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ printf "%s-workers" .Release.Name }}
+  labels:
+    release: {{ .Release.Name }}
+    role: datamigrate-job
+  ownerReferences:
+  {{- if .Values.owner.enabled }}
+    - apiVersion: {{ .Values.owner.apiVersion }}
+      blockOwnerDeletion: {{ .Values.owner.blockOwnerDeletion }}
+      controller: {{ .Values.owner.controller }}
+      kind: {{ .Values.owner.kind }}
+      name: {{ .Values.owner.name }}
+      uid: {{ .Values.owner.uid }}
+  {{- end }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ printf "%s-workers" .Release.Name }}
+  # match the service name
+  serviceName: {{ printf "%s-svc" .Release.Name }}
+  replicas: {{ .Values.datamigrate.parallelism }}
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      labels:
+        app: {{ printf "%s-workers" .Release.Name }}
+    spec:
+      containers:
+        - name: worker
+          # juice fs with openssh server
+          image: {{ required "DataMigrate image should be set" .Values.datamigrate.image }}
+          imagePullPolicy: IfNotPresent
+          command: ["/usr/sbin/sshd", "-D"]
+          env:
+            {{- range .Values.datamigrate.encryptOptions }}
+            - name: {{ .name }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .valueFrom.secretKeyRef.name }}
+                  key: {{ .valueFrom.secretKeyRef.key }}
+            {{- end }}
+          volumeMounts:
+            - mountPath: /root/.ssh
+              name: data-migrate-ssh
+              # use subpath to avoid permissions check problem because the launcher will ssh to workers.
+              # Or user can set 'StrictModes no' in /etc/ssh/sshd_config file to avoid permission check when build the image in dockerfile.
+              subPath: .ssh
+      affinity:
+        # prefer to use different nodes to maximize bandwidth
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - podAffinityTerm:
+                labelSelector:
+                  matchExpressions:
+                    - key: app
+                      operator: In
+                      values:
+                        - {{ printf "%s-workers" .Release.Name }}
+                topologyKey: kubernetes.io/hostname
+              weight: 100
+      volumes:
+        - name: data-migrate-ssh
+          secret:
+            secretName: {{.Values.datamigrate.sshSecretName }}
+            defaultMode: 0600
+            items:
+              - key: ssh-privatekey
```

Comment:
Similar to the job's spec. Does sts need ssh-privatekey?

Reply:
needed as the sts use `ssh localhost` as readinessprobe.